event-sourcing/event-publisher/linked-event-processor/src/main/java/uk/gov/justice/services/eventsourcing/linkedevent/ActiveStreamsProcessor.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.justice.services.eventsourcing.linkedevent;
+
+import uk.gov.justice.services.eventsourcing.repository.jdbc.event.EventJdbcRepositoryFactory;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.eventstream.EventStream;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.eventstream.EventStreamJdbcRepository;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.eventstream.EventStreamJdbcRepositoryFactory;
+import uk.gov.justice.subscription.registry.DefaultEventSourceDefinitionFactory;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import javax.inject.Inject;
+
+public class ActiveStreamsProcessor {
+
+    @Inject
+    private LinkedEventsProcessor linkedEventsProcessor;
+
+    @Inject
+    private EventStreamJdbcRepositoryFactory eventStreamJdbcRepositoryFactory;
+
+    @Inject
+    private EventJdbcRepositoryFactory eventJdbcRepositoryFactory;
+
+    @Inject
+    private DefaultEventSourceDefinitionFactory defaultEventSourceDefinitionFactory;
+
+    public void populateLinkedEvents() {
+
+        final Optional<String> dataSource = defaultEventSourceDefinitionFactory.createDefaultEventSource().getLocation().getDataSource();
+
+        final String jndiDatasource = dataSource.get();
+        final EventStreamJdbcRepository eventStreamJdbcRepository = eventStreamJdbcRepositoryFactory.eventStreamJdbcRepository(jndiDatasource);
+
+        final Stream<EventStream> activeStreams = eventStreamJdbcRepository.findActive();
```

Comment:
this should be a try resource so that the Stream is closed even if an Exception occurs

Reply:
fixed